src/Events.Functions/Clients/EventsClient.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Altinn.Common.AccessTokenClient.Services;
+using Altinn.Platform.Events.Functions.Clients.Interfaces;
+using Altinn.Platform.Events.Functions.Configuration;
+using Altinn.Platform.Events.Functions.Extensions;
+using Altinn.Platform.Events.Functions.Models;
+using Altinn.Platform.Events.Functions.Services.Interfaces;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.Platform.Events.Functions.Clients
+{
+    /// <summary>
+    /// Interface to Events API
+    /// </summary>
+    public class EventsClient : IEventsClient
+    {
+        private readonly ILogger<IEventsClient> _logger;
+
+        private readonly HttpClient _client;
+
+        private readonly IAccessTokenGenerator _accessTokenGenerator;
+
+        private readonly IKeyVaultService _keyVaultService;
+
+        private readonly KeyVaultSettings _keyVaultSettings;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventsClient"/> class.
+        /// </summary>
+        public EventsClient(
+            HttpClient client,
+            IAccessTokenGenerator accessTokenGenerator,
+            IKeyVaultService keyVaultService,
+            IOptions<KeyVaultSettings> keyVaultSettings,
+            ILogger<IEventsClient> logger)
+        {
+            _client = client;
+            _accessTokenGenerator = accessTokenGenerator;
+            _keyVaultService = keyVaultService;
+            _keyVaultSettings = keyVaultSettings.Value;
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventsClient"/> class.
+        /// </summary>
+        public EventsClient(
+            HttpClient httpClient,
+            IAccessTokenGenerator accessTokenGenerator,
+            IKeyVaultService keyVaultService,
+            IOptions<PlatformSettings> eventsConfig,
+            IOptions<KeyVaultSettings> keyVaultSettings,
+            ILogger<IEventsClient> logger)
+        {
+            _client = httpClient;
+            _accessTokenGenerator = accessTokenGenerator;
+            _keyVaultService = keyVaultService;
+            _keyVaultSettings = keyVaultSettings.Value;
+
+            var platformSettings = eventsConfig.Value;
+            _client.BaseAddress = new Uri(platformSettings.ApiEventsEndpoint);
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Generate a fresh access token using the client certificate
+        /// </summary>
+        /// <returns></returns>
+        protected async Task<string> GenerateAccessToken(string issuer, string app)
```

Comment:
for this function in particular issuer and app are always the same. Could maybe be props of the class instead of passing them in for each request?